src/main/java/com/google/sps/servlets/generateServlet.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashSet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns bucket list content */
+@WebServlet("/generate")
+public class generateServlet extends HttpServlet {
+
+  private static final String FILTER_ARRAY = "clicked-array";
+  String JSONclickedArray;
+  HashSet<String> clickedFilters;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get array of clicked filters and convert to ArrayList<String>
+    JSONclickedArray = request.getParameter(FILTER_ARRAY);
+    Gson gson = new Gson();
+    clickedFilters = gson.fromJson(JSONclickedArray, HashSet.class);
```

Comment:
The doPost method is the only place this variable is used, so no need to make this a class variable. In fact, you don't need a local variable at all.  Same for clickedFilters.  If it's not used elsewhere, you should contain the scope of variables to the methods they are used in.
 
```suggestion
    HashSet<String> clickedFilters = gson.fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
```